pkg/schema/error.go

```diff
@@ -155,29 +154,34 @@ type schemaAssertionError struct {
 	hints        []string
 }
 
-type annErr struct {
+type posInfo struct {
 	Pos       string
 	Source    string
 	SkipLines bool
 }
 
-func createAnnInfo(annPosList []*filepos.Position, nodePos *filepos.Position) []annErr {
+func createPosInfo(annPosList []*filepos.Position, nodePos *filepos.Position) []posInfo {
 	sort.SliceStable(annPosList, func(i, j int) bool {
+		if !annPosList[i].IsKnown() {
+			return true
+		}
+		if !annPosList[j].IsKnown() {
+			return false
+		}
 		return annPosList[i].LineNum() < annPosList[j].LineNum()
 	})
 
-	var AnnPositions []annErr
-	for i, p := range annPosList {
-		//if last position in list, compare with node position
-		if i+1 == len(annPosList) {
-			skipLines := !p.IsNextTo(nodePos)
-			AnnPositions = append(AnnPositions, annErr{Pos: p.AsIntString(), Source: p.GetLine(), SkipLines: skipLines})
-		} else {
-			skipLines := !p.IsNextTo(annPosList[i+1])
-			AnnPositions = append(AnnPositions, annErr{Pos: p.AsIntString(), Source: p.GetLine(), SkipLines: skipLines})
+	allPositions := append(annPosList, nodePos)
+	var positionsInfo []posInfo
+	for i, p := range allPositions {
+		//last position in list
+		skipLines := false
+		if i < len(allPositions)-1 {
+			skipLines = !p.IsNextTo(allPositions[i+1])
```

Comment:
nit: improvement in readability?  

```suggestion
		if i > 0 {
			skipLines = !p.IsNextTo(allPositions[i-1])
		}
```

... and swap the order of `...` and source in the template:

```diff
  {{- range .Positions}}
- {{pad "|" .Pos}} {{.Source}}
- {{- if .SkipLines}}
- {{pad "|" ""}} {{"..."}}
- {{- end}}
+ {{- if .SkipLines}}
+ {{pad "|" ""}} {{"..."}}
+ {{- end}}
+ {{pad "|" .Pos}} {{.Source}}
  {{- end}}
```